share/src/main/java/io/zensoft/share/service/Facebook/FacebookPublisherService.java

```diff
@@ -46,50 +52,53 @@ public FacebookPublisherService() throws Exception {
     }
 
     @Deprecated // Only for my Test User. Another method for getting access token of ANY user will be written later.
-    private String getTestUserAccessToken() throws Exception {
+    private String getTestUserAccessToken(){
         Map<String, String> uriVariables = new LinkedHashMap<>();
-        ResponseEntity<Map> map;
+        ResponseEntity<Map> map = null;
         try {
             map = ((FacebookTemplate) facebookApp).getRestTemplate().exchange(
                     "https://graph.facebook.com/" + properties.getProperty("appId") + "/accounts/test-users/",
                     HttpMethod.GET, (HttpEntity<?>) null, Map.class, (Object) uriVariables);
         } catch (Exception e) {
-            throw new Exception(e.getMessage());
+
         }
         String access_token = ((ArrayList<Map<String, Object>>)map.getBody().get("data")).get(0).get("access_token").toString();
         return access_token;
     }
 
-    private String getPageAccessToken () throws Exception {
+    private String getPageAccessToken () {
         Map<String, String> uriVariables = new LinkedHashMap<>();
-        ResponseEntity<Map> map;
+        ResponseEntity<Map> map = null;
         try {
             map = ((FacebookTemplate) facebookUser).getRestTemplate().exchange(
                     "https://graph.facebook.com/" + properties.getProperty("userId") + "/accounts",
                     HttpMethod.GET, (HttpEntity<?>) null, Map.class, (Object) uriVariables);
         } catch (Exception e) {
-            throw new Exception(e.getMessage());
+
         }
         String access_token = ((ArrayList<Map<String, Object>>)map.getBody().get("data")).get(0).get("access_token").toString();
         return access_token;
     }
 
+    @Override
     public VacancyResponse publish(Vacancy vacancy) {
-        PagePostData pagePostData = new PagePostData(properties.getProperty("pageId"));
-        pagePostData.message(getText(vacancy));
-        String postId = facebookUser.pageOperations().post(pagePostData);
+        return publishPhoto(vacancy);
+    }
+
+    @Override
+    public VacancyResponse getInfo(Vacancy vacancy) {
         return null;
     }
 
-    public VacancyResponse publishPhoto(Vacancy vacancy) throws Exception {
+    public VacancyResponse publishPhoto(Vacancy vacancy) {
```

Comment:
did i get it right, if no picture is provided your code stops working when it **catches an error** instead of publishing the post without picture?

Reply:
Conversely it won't stop if the error is catched. The info about the error will be added to response. Code for handling errors has not written yet.